compiler/optimizer/LoopReducer.cpp

```diff
@@ -1088,23 +1088,24 @@ TR_LoopReducer::generateArraycopy(TR_InductionVariable * indVar, TR::Block * loo
    if (arraycopyLoop.getStoreAddress()->getIncrement() < 0)
       //&& (storeAddr->getFirstChild()->getSymbol()->getRegisterMappedSymbol() == loadAddr->getFirstChild()->getSymbol()->getRegisterMappedSymbol()))
       {
-      if (comp()->cg()->getSupportsPrimitiveArrayCopy())
-         arraycopy->setBackwardArrayCopy(true); /* bit available only to primitive arraycopy */
```

Comment:
It looks safe to call `setBackwardArrayCopy(true)` here regardless of `getSupportsPrimitiveAraryCopy`. It does unset `is[RarePath]ForwardArrayCopy`, but that's already not set on this new `arraycopy` node. So the only difference should be the answer to `isBackwardArrayCopy`, which AFAICT is only *really* queried by `ArraycopyTransformation::createMultipleArrayNodes`, which is only called on z where we were already explicitly marking it as backward.

The other uses I found of `isBackwardArrayCopy` are just in assertions in related `arraycopy` flag setter methods, and code that copies properties from one `arraycopy` node to another.